pkg/ride/environment.go

```diff
@@ -1110,15 +1017,15 @@ func NewEnvironmentWithWrappedState(
 	isProtobufTransaction bool,
 	rootScriptLibVersion ast.LibraryVersion,
 ) (*EvaluationEnvironment, error) {
-	recipient := proto.NewRecipientFromAddress(proto.WavesAddress(env.th.(rideAddress)))
-
+	recipient := proto.WavesAddress(env.th.(rideAddress))
 	st := newWrappedState(env, rootScriptLibVersion)
-	for _, payment := range payments {
+	for i, payment := range payments {
 		var (
 			senderBalance uint64
 			err           error
 			callerRcp     = proto.NewRecipientFromAddress(sender)
 		)
+		// TODO: Move validation after application
```

Comment:
Why?

Reply:
We can apply all payments and later check all balances. No need to check them as we go.